src/bin/decode.rs

```diff
@@ -92,12 +100,40 @@ fn main() {
         root: &builder.node_name("Script"),
     };
     let spec = builder.into_spec(spec_options);
+    let printer = Shift::new();
     let source = printer.to_source(&spec, &json)
         .expect("Could not pretty-print");
 
     progress!(quiet, "Writing.");
-    let mut dest = File::create(dest_path)
-        .expect("Could not create destination file");
-    dest.write(source.as_bytes())
-        .expect("Could not write destination file");
+    match options.dest_path {
+        Some(path) => {
+            let mut dest = File::create(path)
+                .expect("Could not create destination file");
+            dest.write(source.as_bytes())
+                .expect("Could not write destination file");
+        }
+        None => {
+            stdout().write(source.as_bytes())
+                .expect("Could not write destination file");
+        }
+    }
+}
+
+fn parse_tree<R: Read + Seek>(get_stream: &Fn() -> R, options: &Options) -> binjs::specialized::es6::ast::Script
+{
+    if !options.quiet {
```

Comment:
Note for later: we'll eventually need to implement a way to sniff various formats a little bit more nicely.